pkg/resource/alias/hooks.go

```diff
@@ -80,20 +82,29 @@ func (rm *resourceManager) updateProvisionedConcurrency(
 	exit := rlog.Trace("rm.updateProvisionedConcurrency")
 	defer exit(err)
 
+	// Check if the user deleted the 'ProvisionedConcurrency' configuration
+	// If yes, delete ProvisionedConcurrencyConfig
+	if desired.ko.Spec.ProvisionedConcurrencyConfig == nil {
+		input_delete := &svcsdk.DeleteProvisionedConcurrencyConfigInput{
+			FunctionName: aws.String(*desired.ko.Spec.FunctionName),
+			Qualifier:    aws.String(*desired.ko.Spec.Name),
+		}
+		_, err = rm.sdkapi.DeleteProvisionedConcurrencyConfigWithContext(ctx, input_delete)
+		rm.metrics.RecordAPICall("DELETE", "DeleteProvisionedConcurrency", err)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
 	dspec := desired.ko.Spec
 	input := &svcsdk.PutProvisionedConcurrencyConfigInput{
 		FunctionName: aws.String(*dspec.FunctionName),
 		Qualifier:    aws.String(*dspec.Name),
 	}
 
-	if desired.ko.Spec.ProvisionedConcurrencyConfig != nil {
-		if desired.ko.Spec.ProvisionedConcurrencyConfig.ProvisionedConcurrentExecutions != nil {
-			input.ProvisionedConcurrentExecutions = aws.Int64(*desired.ko.Spec.ProvisionedConcurrencyConfig.ProvisionedConcurrentExecutions)
-		} else {
-			input.ProvisionedConcurrentExecutions = aws.Int64(0)
-		}
-	} else {
-		input.ProvisionedConcurrentExecutions = aws.Int64(0)
+	if dspec.ProvisionedConcurrencyConfig.ProvisionedConcurrentExecutions != nil {
```

Comment:
What if the user sets `dspec.ProvisionedConcurrencyConfig.ProvisionedConcurrentExecutions` to nil ? is that equivalent to setting the value to 0 or 1, or maybe delete it fully ?

Reply:
If the user sets ProvisionedConcurrency to 1, they will get the error as ProvisionedConcurrency must be greater than 1. For nil, I added the condition along with nil checking for `ProvisionedConcurrency` in the beginning of the `updateProvisionedConcurrency` function.

Reply:
Since you added `if desired.ko.Spec.ProvisionedConcurrencyConfig == nil || desired.ko.Spec.ProvisionedConcurrencyConfig.ProvisionedConcurrentExecutions == nil {` can we remove this `if` statement now?

Reply:
Done